ensure marker symbols are drawn in the appropriate pane

jgravois opened the pull request:
@pbastia can you provide a little clarification about what client side developers need to do to benefit from your changes in #112 when you have a moment?

i haven't had success using the approach below (testing against leaflet `1.0.0-rc.1`).
```js
map.createPane('custom');

// circle marker symbol
L.esri.featureLayer({
  url: 'http://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer/0',
  pane: 'custom'
}).addTo(map);

/* 
also tried picture marker symbols
https://services.arcgis.com/rOo16HdIMeOBI4Mb/arcgis/rest/services/Trimet%20Transit%20Stops%20In%20Portland%20Neighborhoods/FeatureServer/0

and this layer which includes diamond markers, among others...
http://sampleserver6.arcgisonline.com/arcgis/rest/services/Military/MapServer/3
*/
```

pbastia commented:
@jgravois I see the issue here
This is making use of the 'leaflet-shape-markers' library. Even the API exposes an 'options' field to create markers, it seems like these options are not passed properly.
I'll do some digging when I have more time.

In my particular case I was using the 'esriPMS' layer type, and it worked as expected.

jgravois commented:
i feel like i *must* be doing something wrong, but even when i check out your own [branch](https://github.com/pbastia/esri-leaflet-renderers), esriPMS symbols like the one in [this service](http://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer/3) are drawing in Leaflet's default marker pane in my own test application.

jgravois commented:
its as likely as anything that i missed something simple in @pbastia's solution, but since i didn't hear back i made a few small tweaks which allow users to pass their own `pane`s directly within featureLayer constructors and ensure that the various point symbols are all drawn in the appropriate place.

* i removed my own previous, totally irrelevant tests (not sure what i was thinking when i added them)
* i created a new visual one page app in the `spec/` directory for QA down the line
* moved all urls in existing `comparisons.html app` to use SSL
* tweaked code so that panes are passed through to `Renderer.pointToLayer`

jgravois commented:
@kneemer can you spare a few minutes to review this PR too?  if you're happy with it, i'll go ahead and tag a release for the fine young gentleman that's tackling webmap support in esri-leaflet.

ynunokawa/L.esri.WebMap#17

kneemer commented:
Already checking it out 😄 .  Thanks for fixing `comparisons.html` was just getting ready to make a PR for it.

kneemer commented on spec/custom-pane.html:
```diff
@@ -0,0 +1,56 @@
+
+      // misc marker symbols
+      L.esri.featureLayer({
+        url: 'http://sampleserver6.arcgisonline.com/arcgis/rest/services/CommunityAddressing/MapServer/0'// ,
```
Is this the service that doesn't render noted in #115?

jgravois replied:
yeah, i got sidetracked, but i can/will definitely stick a valid url back in

kneemer commented on src/Renderers/Renderer.js:
```diff
@@ -64,7 +64,7 @@ export var Renderer = L.Class.extend({
     var sym = this._getSymbol(geojson);
     if (sym && sym.pointToLayer) {
-      return sym.pointToLayer(geojson, latlng, this._visualVariables, options);
+      return sym.pointToLayer(geojson, latlng, this._visualVariables, this.options);
```
Since you are not using the options parameter you should probably remove it to reduce confusion.  Also, this change would be specific to panes and not allow for additional options which is fine I think but not what the contributor was trying to accomplish.  I think they were trying to let all options through.

jgravois replied:
agreed.  

i honestly can't figure out how his own implementation could have worked at all though.

jgravois commented:
@kneemer, i laid down one more commit based on your suggestions.  do you feel like this is ready to merge?

kneemer commented:
Looks good to me.
